### PR TITLE
Raise error if ru specific instrument uploaded

### PR DESCRIPTION
**Pull request by insacuri**
# What and why?

# How to test?

# Trello

### CODE REVIEW

File: tests/controllers/test_service_helper.py [hunk 1]
```diff
@@ -46,8 +46,8 @@ def test_service_request_http_error(self):
                     service=SERVICE, endpoint="surveys", search_value="41320b22-b425-4fba-a90e-718898f718ce"
                 )
 
-        self.assertEquals(["survey-service returned a HTTPError"], exception.exception.errors)
-        self.assertEquals(500, exception.exception.status_code)
+        self.assertEqual(["survey-service returned a HTTPError"], exception.exception.errors)
```
Comment (LJBabbage) [inline tests/controllers/test_service_helper.py:49]: Thanks

File: application/controllers/collection_instrument.py [hunk 2]
```diff
@@ -96,6 +95,9 @@ def upload_seft_to_bucket(self, exercise_id, file, ru_ref=None, classifiers=None
         survey = self._find_or_create_survey_from_exercise_id(exercise_id, session)
         survey_id = survey.survey_id
         survey_service_details = get_survey_details(survey_id)
+        if survey_service_details["surveyMode"] == "EQ_AND_SEFT" and ru_ref is not None:
+            raise RasError("Can't upload an reporting unit specific instrument for an EQ_AND_SEFT survey", 400)
```
Comment (LJBabbage) [inline application/controllers/collection_instrument.py:99]: "a" instead or "an"

**Review by LJBabbage — APPROVED** [review]
Works as expected, the tiniest of grammar changes

File: application/controllers/collection_instrument.py (hunk 2 above)
Comment (insacuri) [reply application/controllers/collection_instrument.py:99]: Doh!  I went back and fourth on a few different wordings and missed that.  Thanks for catching it ^^

**Pull request merged**